dockerfile/azure-nvrt-nhc.dockerfile

```diff
@@ -10,12 +10,15 @@ ENV AOCC_VERSION=4.0.0_1
 ENV PERF_TEST_VERSION=23.10.0
 ENV NV_BANDWIDTH_VERSION=0.4
 ENV NCCL_VERSION=2.19.3-1
-ENV HPCX_VERSION=v2.16
+ENV OPEN_MPI_VERSION=4.1.5
```

Comment:
Is this the OMPI that HPCX 2.16 is based on?

Reply:
Images are moving to OMPI 5.0.2

Reply:
I was referencing what was currently in the main repo of HPC images. Since this is what I tested with, I think I will keep it the same for now while we wait for clearance to use HPCx.